lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -670,6 +670,564 @@ class AuthList
+          @$section.find('.problem_specific').addClass("active")
+
+  get_list: (cb)->
+    $.ajax
+      dataType: 'json'
+      url: @$list_endpoint
+      data: rolename: 'instructor'
+      success: (data) -> cb? null, data['data']
+      error: std_ajax_err ->
+        cb? gettext("Error fetching problem or section data")
+
+  # load section/problem data
+  load_list: ->
+    @get_list (error, section_list) =>
+      # abort on error
+      return @show_errors error unless error is null
+      _.each section_list, (section) =>
+        @add_row(section, "section")
+        _.each section.sub, (subsection) =>
+          @add_row(subsection , "subsection")
+
+  add_row: (node, useClass) ->
+    @idArr = [node.block_type, node.block_id]
+    @idSt = @idArr.join("/")
+    @toDisplay = node.display_name
+    if node.parents
+      @toDisplay = [node.parents,@toDisplay].join("<>")
+    #indenting subsections with dashes for readability
+    if useClass == "subsection"
+      @toDisplay = "---" + @toDisplay
+    if @toDisplay.length > 50
+      #displaying the last n characters
+      @toDisplay = "..." + @toDisplay.substring(@toDisplay.length-60, @toDisplay.length)
+    @$list_selector.append $ '<option/>',
+            text: @toDisplay
+            class: useClass
+            id : @idSt
+
+  set_cell: (text, colNumber, cellid) ->
+    rows = $("#queryTableBody")[0].rows
+    cell = rows[rows.length - 1].children[colNumber]
+    if cell
+      cell.innerHTML = text
+      if cellid != ""
+        cell.id = cellid
+
+class EmailWidget
+  constructor:  (@email_lists, @$section, @$email_list_containers)  ->
+    @$query_endpoint = $(".email-lists-management").data('query-endpoint')
+    @$total_endpoint = $(".email-lists-management")
+      .data('total-endpoint')
+    @$temp_queries_endpoint = $(".email-lists-management")
+      .data('temp-queries-endpoint')
+    @$delete_saved_endpoint = $(".email-lists-management")
+      .data('delete-saved-endpoint')
+    @$delete_temp_endpoint = $(".email-lists-management")
+      .data('delete-temp-endpoint')
+    @$delete_bulk_temp_endpoint = $(".email-lists-management")
+      .data('delete-bulk-temp-endpoint')
+    for email_list in @email_lists
+      email_list.$container.addClass 'active'
+
+    @$get_est_btn = @$section.find("input[name='getest']'")
+    @$get_est_btn.click () =>
+      @reload_estimated()
+
+    @$startover_btn = @$section.find("input[name='startover']'")
+    @$startover_btn.click () =>
+      @delete_temporary()
+      $("#queryTableBody tr").remove()
+      @reload_estimated()
+
+    @$save_query_btn = @$section.find("input[name='savequery']'")
+    @$save_query_btn.click () =>
+      @send_save_query()
+
+    @$email_csv_btn = @$section.find("input[name='getcsv']'")
+    @$email_csv_btn.click () =>
+      sending_query = []
+      tab = $("#queryTableBody")
+      rows = tab.find("tr")
+      _.each rows, (row) ->
+        sending_query.push(row.getAttribute('query'))
```

Comment:
Since we're creating a new array (`sending_query`) based on each element of another array (`rows`), it looks like `_.map(...)` would be better than `_.each` here.